lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
 class Tree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :age, :height, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    self.age += 1
+    self.height += 1
+    self.add_apples
+    if age >= 20
+      self.alive = false
+    end
   end
 
   def add_apples
+    3.times do self.apples << Apple.new(rand(4..20))
+    end
   end
 
   def any_apples?
+    apples.length > 0 ? true : false
   end
 
   def pick_an_apple!
-    raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    unless any_apples?
```

Comment:
Try to avoid `unless`/`else`. Its easir for the next developer to read if you make it `if`/`else` and flip the two branches.